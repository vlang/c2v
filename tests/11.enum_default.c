enum myEnum {
    A,
    B,
    C
};

typedef enum {
    D,
    E,
    F
} myAnotherEnum;

typedef enum myStrangeEnum {
    G,
    H,
    I
} myStrangeEnum;

enum { J = 1 };

void enum_func(enum myEnum a) {
}

void enum_func_const(const enum myEnum a) {
}

int main() {
    enum myEnum myEnumVar = A;
    myAnotherEnum myEnumVar2 = D;
    myStrangeEnum myEnumVar3 = G;
    int myIntVar = J;
    enum_func(myEnumVar);
    enum_func_const(myEnumVar);
    return 0;
}
