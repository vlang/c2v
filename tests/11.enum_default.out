[translated]
module main

enum MyEnum {
	a
	b
	c
}

enum MyAnotherEnum {
	d
	e
	f
}

enum MyStrangeEnum {
	g
	h
	i
}

const ( // empty enum
	j = 1
)

fn enum_func(a MyEnum) {
}

fn enum_func_const(a MyEnum) {
}

fn main() {
	myenumvar := MyEnum.a
	myenumvar2 := MyAnotherEnum.d
	myenumvar3 := MyStrangeEnum.g
	myintvar := j
	enum_func(myenumvar)
	enum_func_const(myenumvar)
	return
}
